source/darwin/secure_transport_tls_channel_handler.c

```diff
@@ -656,7 +657,8 @@ static int s_process_read_message(
         if (slot->adj_right) {
             if (aws_channel_slot_send_message(slot, outgoing_read_message, AWS_CHANNEL_DIR_READ)) {
                 aws_mem_release(outgoing_read_message->allocator, outgoing_read_message);
-                return AWS_OP_ERR;
+                /* incoming message was pushed to the input_queue, so this handler owns it now */
+                return AWS_OP_SUCCESS;
```

Comment:
Do we need to explicitly shutdown?

Reply:
Good question, we probably do, let me check